sdks/js/packages/core/react/components/organization/plans/pricing-column.tsx

```diff
@@ -0,0 +1,354 @@
+
+interface FeaturesListProps {
+  features: string[];
+  plan: IntervalPricingWithPlan;
+}
+
+const FeaturesList = ({ features, plan }: FeaturesListProps) => {
+  return features.map(feature => {
+    const planFeature = _.get(plan.features, feature, {
+      metadata: {}
+    });
+    const productMetaDataFeatureValues = plan.productNames
+      .map(name => _.get(planFeature.metadata, name))
+      .filter(value => value !== undefined);
+    // piciking the first value for feature metadata, in case of muliple products in a plan, there can be multiple metadata values.
+    const value = productMetaDataFeatureValues[0];
+    const isAvailable = value?.toLowerCase() === 'true';
+    return (
+      <Flex
+        key={feature + '-' + plan.planId}
+        align={'center'}
+        justify={'start'}
+        className={plansStyles.featureCell}
+      >
+        {isAvailable ? (
+          <Image
+            // @ts-ignore
+            src={checkCircle}
+            alt="checked"
+          />
+        ) : value ? (
+          <Text>{value}</Text>
+        ) : (
+          <Text>-</Text>
+        )}
+      </Flex>
+    );
+  });
+};
+
+interface PlanIntervalsProps {
+  planIntervals: IntervalKeys[];
+  selectedInterval: IntervalKeys;
+  onIntervalChange: (i: IntervalKeys) => void;
+}
+
+const PlanIntervals = ({
+  planIntervals,
+  selectedInterval,
+  onIntervalChange
+}: PlanIntervalsProps) => {
+  return planIntervals.length > 1 ? (
+    <ToggleGroup
+      className={plansStyles.plansIntervalList}
+      value={selectedInterval}
+      onValueChange={onIntervalChange}
+    >
+      {planIntervals.map(key => (
+        <ToggleGroup.Item
+          value={key}
+          key={key}
+          className={plansStyles.plansIntervalListItem}
+        >
+          <Text className={plansStyles.plansIntervalListItemText}>
+            {IntervalLabelMap[key]}
+          </Text>
+        </ToggleGroup.Item>
+      ))}
+    </ToggleGroup>
+  ) : null;
+};
+
+interface TrialLinkProps {
+  planIds: string[];
+  isUpgrade: boolean;
+  planHasTrial: boolean;
+  onButtonClick: () => void;
+  disabled: boolean;
+  dateFormat: string;
+}
+
+const TrialLink = function TrialLink({
+  disabled,
+  planIds,
+  isUpgrade,
+  planHasTrial,
+  dateFormat,
+  onButtonClick = () => {}
+}: TrialLinkProps) {
+  const {
+    isTrailCheckLoading,
+    hasAlreadyTrailed,
+    checkAlreadyTrialed,
+    trailSubscription
+  } = usePlans();
+
+  useEffect(() => {
+    if (planHasTrial) {
+      checkAlreadyTrialed(planIds);
+    }
+  }, [checkAlreadyTrialed, planHasTrial, planIds]);
+
+  const trailEndDate = planIds.includes(trailSubscription?.plan_id || '')
+    ? dayjs(trailSubscription?.trial_ends_at).format(dateFormat)
+    : '';
+
+  const showButton = isUpgrade && !hasAlreadyTrailed && planHasTrial;
+  return (
+    <Flex
+      className={plansStyles.trialWrapper}
+      justify={'center'}
+      align={'center'}
+    >
+      {isTrailCheckLoading ? (
```

Comment:
Is there a more readable way we can put this? Nested ternaries can get a bit difficult to understand later on

Reply:
we can change something when we do refactoring in future